cpp/memilio/math/integrator.h

```diff
@@ -38,6 +39,19 @@ template <typename FP = double>
 class IntegratorCore
 {
 public:
+    /**
+     * @brief Initialize an IntegratorCore.
+     * Step size bounds are needed for adaptive integrators, see the step method for more detail.
+     * Fixed size steppers ignore those bounds and may use the default constructor for FP.
```

Comment:
My idea was to nudge users to look at this documentation by not having a default constructor, and especially avoid `dt_min=dt_max=0` for adaptive steppers. We can set different defaults, but that potentially could also cause issues. And it is not that much more difficult to use.

I am not completely against adding a default ctor, but I think not having one might be better.

Reply:
When using non-adaptive methods, IMO it is better to have a default constructor, because dt_min and dt_max make no sense there.
Is there anything wrong with dt_min=0 and dt_max="FP_MAX"? In any case, you should read the documentation if you use the class..

Reply:
> you should read the documentation if you use the class..

Yeah, that would be a nice change...

> Is there anything wrong with dt_min=0 and dt_max="FP_MAX"?

Technically any choice can have problems. dt_min = 0 allows for infinite loops, and "FP_MIN" is not better on a human time scale. Any "larger" value may cause loss of precision.
But the main point is to move that choice away from the interface (IntegratorCore) and to the actual implementations. 

Maybe it would be better to split IntegratorCore into adaptive and non-adaptive? Then we could avoid the issue of "just ignore dt_min/_max" and simplify the documentation.

Reply:
I understand your arguments against default values. I dont like the idea of having two classes. It is an advantage, that we have just one class for all solvers ATM. Under these conditions, I think it's fine not to have a default constructor.